L1Trigger/L1TMuonEndCap/src/helper.h

```diff
@@ -82,6 +82,36 @@ namespace {
     }
   }
 
+  // Check type for map of vector
+  template<typename>
+  struct is_map_of_vectors : public std::false_type { };
+
+  template<typename T1, typename T2>
+  struct is_map_of_vectors<std::map<T1, std::vector<T2> > > : public std::true_type { };
+
+  // Merge a map of vectors (map1) into another map of vectors (map2)
+  template<typename Map>
+  void merge_map_into_map(Map& map1, Map& map2) {
+    // This is customized for maps of containers.
+    typedef typename Map::iterator Iterator;
+    typedef typename Map::mapped_type Container;
+
+    Iterator first = map1.begin();
+    Iterator last = map1.end();
+    for (; first != last; ++first) {
```

Comment:
This will probably have to be addressed when PRed into CMSSW.
Using range-based for loops is highly encouraged where possible, 
and this is one of the cases.
Will merge into l1t-integration for now.